Feature 836 rtd release doc

jprestop opened the pull request:
## Pull Request Testing ##

Updating link references from dtcenter.github.io to metplus.readthedocs.io are also included in this pull request.

- [x] Describe testing already performed for these changes:</br>

Reviewed the modified sections:

1. https://metplus.readthedocs.io/en/feature_836_rtd_release_doc/Release_Guide/metplus_official.html#update-the-documentation-on-the-web (This same text exists in the official release sections for METcalcpy, METplotpy, and MET - other components do not yet have documentation set up; There is no documentation section needed for bugfix or development releases)
2. https://metplus.readthedocs.io/en/feature_836_rtd_release_doc/Contributors_Guide/documentation.html
3. https://metplus.readthedocs.io/en/feature_836_rtd_release_doc/Contributors_Guide/add_use_case.html#accessing-the-documentation

- [x] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:</br>

Review the files changed and point out anything that may be unclear or needs additional documentation.  Review the web links above to ensure they look ok to you.

- [x] Do these changes include sufficient documentation updates, ensuring that no errors or warnings exist in the build of the documentation? **[Yes]**. I do not see a warnings file here: https://github.com/dtcenter/METplus/actions/runs/723531239

- [x] Do these changes include sufficient testing updates? **[Yes]** None needed

- [x] Will this PR result in changes to the test suite? **[No]**</br>
If **yes**, describe the new output and/or changes to the existing output:</br>

## Pull Request Checklist ##
See the [METplus Workflow](https://dtcenter.github.io/METplus/Contributors_Guide/github_workflow.html) for details.
- [x] Complete the PR definition above.
- [x] Ensure the PR title matches the feature or bugfix branch name.
- [x] Define the PR metadata, as permissions allow.
Select: **Reviewer(s)**, **Project(s)**, and **Milestone**
- [x] After submitting the PR, select **Linked Issues** with the original issue number.
- [ ] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [ ] Close the linked issue and delete your feature or bugfix branch from GitHub.

jprestop committed:
Added section for ReadTheDocs, added basic info about the Release Guide and Verification Datasets Guide, but more information on those needs to be added later

jprestop committed:
Added information about the various versions available in the documentation

jprestop committed:
Updated the ReadTheDocs section with information on hidden branches and how to access them; Modified dtcenter.github.io references

jprestop committed:
Added section to add_use_case.rst about Accessing the Documentation and modified documentation.rst to indicate the docs can still be built manually

jprestop committed:
Modified format of list

jprestop committed:
Modified formatting

jprestop committed:
Changed GitHub Pages links to Read the Docs links

jprestop committed:
Modified formatting

jprestop committed:
Fixed typo

jprestop committed:
Updated Release Guide for Read The Docs

georgemccabe reviewed (changes requested):
These changes mostly look good. It is nice to be able to delete a lot of the files for steps that are no longer needed!

I did notice that there is out-of-date info (paths to use case documentation files and location of .py doc files) in the Documentation page of the Contributor's Guide that were not from these changes. The info is duplicated in the Add Use Case chapter where the info has been updated (where to add use case documentation files, list of model applications, etc). We should figure out where this info should go and reference it from the other page so we don't run into this issue again.

Other than that, the content looks good!

jprestop commented:
I added to information to https://metplus.readthedocs.io/en/feature_836_rtd_release_doc/Contributors_Guide/documentation.html#adding-new-documentation, which appeared to be an overview.  I noticed that it listed the User’s Guide and Contributor’s Guide, but not the Release Guide or the Verification Datasets Guide, so I added information about those.  I did not review these sections for duplicate information. It appeared that this section was out of date since it didn’t include information about the Release Guide or Verification Datasets Guide.  Maybe a separate issue should be created to check for duplicate information in these sections?

jprestop committed:
Took out duplicate data and consolidated sections

jprestop commented:
I went ahead and cleaned it up now.  Once I started to make some corrections, I realize it would be easier to just consolidate the info now.

georgemccabe commented on docs/Contributors_Guide/add_use_case.rst:
```diff
@@ -53,8 +66,19 @@ Use Case Content
+use case should have the following:
+
+* A METplus configuration file where the MET tool name follows PascalCase,
+  e.g. GridStat.conf or ASCII2NC.conf.  This file is a hybrid RST and Python
```
```suggestion
  e.g. GridStat.conf or ASCII2NC.conf.
```

georgemccabe commented on docs/Contributors_Guide/add_use_case.rst:
```diff
@@ -53,8 +66,19 @@ Use Case Content
+
+* A METplus configuration file where the MET tool name follows PascalCase,
+  e.g. GridStat.conf or ASCII2NC.conf.  This file is a hybrid RST and Python
+  file.  If the use case uses a Python embedding script, it should be
```
```suggestion
  If the use case uses a Python embedding script, it should be
```

georgemccabe commented on docs/Contributors_Guide/add_use_case.rst:
```diff
@@ -53,8 +66,19 @@ Use Case Content
+  indicasted in the by adding "_python_embedding" to the MET tool name.
+  e.g. GridStat_python_embedding.conf
+
+If createing a new model applications use case, in the category sub-directory
```
```suggestion
If creating a new model applications use case, in the category sub-directory
```

georgemccabe reviewed (approved):
The updates look good to me. Thanks for making them! I approve.

jprestop committed:
Update docs/Contributors_Guide/add_use_case.rst

Removed sentence

Co-authored-by: George McCabe <23407799+georgemccabe@users.noreply.github.com>

jprestop committed:
Update docs/Contributors_Guide/add_use_case.rst

Fixed typo

Co-authored-by: George McCabe <23407799+georgemccabe@users.noreply.github.com>

jprestop committed:
Update docs/Contributors_Guide/add_use_case.rst

Removed end of previously removed sentence.

Co-authored-by: George McCabe <23407799+georgemccabe@users.noreply.github.com>

jprestop commented:
Thanks for checking and for noticing the duplication.
